dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/JSPExtensionFactory.java

```diff
@@ -598,7 +608,12 @@ public static synchronized GlobalTagLibraryCache getGlobalTagLibraryCache() {
      */
     @Reference(cardinality=ReferenceCardinality.MULTIPLE, policy=ReferencePolicy.DYNAMIC)
     protected void setGlobalTagLibConfig(GlobalTagLibConfig globalTagLibConfig) {
-            getGlobalTagLibraryCache().addGlobalTagLibConfig(globalTagLibConfig);
+            // Wait for pages version to be loaded first before calling getGlobalTagLibraryCache
+            if(!isPagesVersionLoaded){
+                storedGlobalTagLibConfigs.add(globalTagLibConfig);
+            } else {
+                getGlobalTagLibraryCache().addGlobalTagLibConfig(globalTagLibConfig);
+            }
```

Comment:
When `addGlobalTagLibConfig` is called,  it eventually relies on the loaded pages version. However, it's not yet set  since `setGlobalTagLibConfig` is called before `setVersion`.   We cache the `globalTagLibConfig` argument and then call addGlobalTagLibConfig later in the `setVersion` call. 

I spoke to Tom Watson, and  we aren't aware of a way to change the order here.